src/signer/canister/tests/it/utils/test_environment.rs

```diff
@@ -0,0 +1,199 @@
+                .with_wasm(&dfx_wasm_path("cycles_ledger"))
+                .with_arg(
+                    encode_one(LedgerArgs::Init(LedgerInitArgs {
+                        index_id: None,
+                        max_blocks_per_request: 999,
+                    }))
+                    .expect("Failed to encode ledger init arg"),
+                )
+                .deploy_to(pic.clone()),
+        );
+        let signer = SignerPic::from(
+            PicCanisterBuilder::default()
+                .with_wasm(&cargo_wasm_path("signer"))
+                .with_arg(
+                    encode_one(Arg::Init(InitArg {
+                        ecdsa_key_name: format!("test_key_1"),
+                        ic_root_key_der: None,
+                        cycles_ledger: None,
+                    }))
+                    .unwrap(),
+                )
+                .deploy_to(pic.clone()),
+        );
+        let user = Principal::from_text(CALLER).unwrap();
+        let user2 =
+            Principal::from_text("jwhyn-xieqy-drmun-h7uci-jzycw-vnqhj-s62vl-4upsg-cmub3-vakaq-rqe")
+                .unwrap();
+        let users = [
+            Principal::from_text("s2xin-cwqnw-sjvht-gp553-an54g-2rhlc-z4c5d-xz5iq-irnbi-sadik-qae")
+                .unwrap(),
+            Principal::from_text("dmvof-2tilt-3xmvh-c7tbj-n3whk-k2i6b-2s2ge-xoo3d-wjuw3-ijpuw-eae")
+                .unwrap(),
+            Principal::from_text("kjerd-nj73t-u3hhp-jcj4d-g7w56-qlrvb-gguta-45yve-336zs-sunxa-zqe")
+                .unwrap(),
+            Principal::from_text("zxhav-yshtx-vhzs2-nvuu3-jrq66-bidn2-put3y-ulwcf-2gb2o-ykfco-sae")
+                .unwrap(),
+            Principal::from_text("nggqm-p5ozz-i5hfv-bejmq-2gtow-4dtqw-vjatn-4b4yw-s5mzs-i46su-6ae")
+                .unwrap(),
+        ];
+        let unauthorized_user =
+            Principal::from_text("rg3gz-22tjp-jh7hl-migkq-vb7in-i2ylc-6umlc-dtbug-v6jgc-uo24d-nqe")
+                .unwrap();
+        let cycles_depositor = PicCanisterBuilder::default()
+            .with_wasm(&dfx_wasm_path("cycles_depositor"))
+            .with_controllers(vec![user])
+            .with_arg(
+                encode_one(cycles_depositor::InitArg {
+                    ledger_id: ledger.canister_id,
+                })
+                .unwrap(),
+            )
+            .deploy_to(pic.clone())
+            .into();
+
+        let ans = Self {
+            pic,
+            signer,
+            ledger,
+            user,
+            user2,
+            users,
+            unauthorized_user,
+            cycles_depositor,
+        };
+        ans.fund_user(Self::USER_INITIAL_BALANCE);
+        ans
+    }
+}
+impl TestSetup {
+    /// The user's initial balance.
+    pub const USER_INITIAL_BALANCE: u128 = 100_000_000_000;
+    /// Deposit 100 * the ledger fee in the user's ledger wallet. That should be enough to be getting on with.
+    pub fn fund_user(&self, megasquigs: u128) {
```

Comment:
What is a `megasquigs`?

Reply:
A British colloquial term for "lots of money".  I'll replace it with something like `cycles`.

Reply:
Renamed.